### DISCUSSION TITLE
API for file listing?

### DISCUSSION THREAD

**Discussion by hzhoanglee** [spencerwooo/onedrive-vercel-index#311] [Q&A] [answered]
This project is an amazing project that perfectly suit my need.
But I'm wondering if this project have any API that list all files/folder with information like size, raw link, etc in json for the use of integration to another application.
With 2000 files, I don't think I want to copy and paste raw links for each file.
Thanks.

**Comment by spencerwooo** [comment, marked as answer]
We use `/api?path=<path_to_file_or_folder>` internally. If you want you can try this, though I believe you would need to deal with CORS yourself.

![Screenshot 2022-01-26 at 10 33 15](https://user-images.githubusercontent.com/32114380/151094309-556f8c4b-dfc6-4842-b3b1-c8b7334b66be.png)

**Reply by hzhoanglee** [reply]
Thanks a lot, that's exactly what I'm looking for. Just need some simple task to get permanent raw links but it's alright.

**Comment by spencerwooo** [comment]
> With 2000 files, I don't think I want to copy and paste raw links for each file.

If you need raw links only, you simply construct links like:

```
/api?path=<path_to_file>&raw=true
```

Such as:

```
https://drive.spencerwoo.com/api?path=/%F0%9F%A5%9F%20Some%20test%20files/Previews/image/minion.png&raw=true
```

To get 302 redirects to the raw OneDrive file links. I don't think you need API queries at all.
